cli/src/config.rs

```diff
@@ -170,27 +170,27 @@ pub fn read_configuration(path: &Path) -> Result<Config> {
     Ok(config)
 }
 
-pub fn find_project_conf(starting_directory: &Path) -> Option<PathBuf> {
-    let mut path: PathBuf = starting_directory.into();
-    let mut attempts = 0;
-    const MAX_DEPTH: u8 = 32;
-
-    loop {
-        let search_path = path.join(PROJ_CONF_FILE);
-        if search_path.is_file() {
-            return Some(search_path);
+pub fn find_project_conf(
+    starting_directory: impl AsRef<Path>,
+    recurse_upwards: bool,
```

Comment:
While we could have checked if the `find_project_conf` returned Path's parent is equal to the current directory, that would have meant traversing the entire directory tree just to discard it. So I've decided to go for the simpler solution and just added an option to skip the traversal.